### ISSUE TITLE
Allow enforce coding standards on Sonar

### ISSUE THREAD

**Issue by GalateaRaj** [GalateaRaj/fuse-starter-java#175]
Issue Description
--
We want to ensure all FUSE developers are following FUSE coding standards (e.g. final modifiers on method params, using a certain import order, using certain spacing, putting curly brackets in the right place).  We can achieve that in two parts:

1.  Providing IDE tools to help implement our standards.  These are stored here for Eclipse:  https://github.com/GalateaRaj/fuse-settings/tree/master/eclipse

2.  Ensuring that our CI process enforces these standards using tools such as checkstlye, sonar, other things I don't know about

***We can't configure sonar at the moment. This issue will cover allowing us to configure Sonar as we wish, then the enforcing of all our standards (and the debating/deciding of those standards) will be covered under issue #176 ***

Design
--
[Description of the root cause for the issue and how to resolve it. Should be complete before Review is approved]

1. Create a new 'Galatea' sonar organization.
2. Migrate sonar analysis over to new organization
2.1 This requires a change to the 'key' for the current fuse sonar project
2.2 And we'll update the 'key' for the new org to be org.galatea:fuse-starter-java
3. Update the sonar profile to align with our standards (This will be covered under Issue #176)
4. Update any documentation we have around configuring Sonar.

Test Evidence
--
[include relevant tests and output. Should be complete before Review is approved]
1. Confirm Jenkins feature/PR builds run under the new Galatea org
![image](https://user-images.githubusercontent.com/23705265/37211142-f657bb84-2378-11e8-80ff-c591d5415194.png)

https://sonarcloud.io/dashboard?id=org.galatea%3Afuse-starter-java
![image](https://user-images.githubusercontent.com/23705265/37211265-5fc04e88-2379-11e8-88d7-d4cdf12de741.png)

Validation in Develop
-- 
[Proof issue is resolved in Develop. Should be complete before Issue is closed]
1. Confirm Jenkins Develop build runs under new Galatea org.
![image](https://user-images.githubusercontent.com/23705265/37212082-0cddf85c-237c-11e8-8d8d-db597fba01de.png)
![image](https://user-images.githubusercontent.com/23705265/37212141-34905796-237c-11e8-9d16-f58a25cfe817.png)

2. Updated documentation:
![image](https://user-images.githubusercontent.com/23705265/37212966-a95ec790-237e-11e8-8081-26fe039d0c43.png)
And it works:
![image](https://user-images.githubusercontent.com/23705265/37212984-bbb39664-237e-11e8-94f9-88ffbc982315.png)

**Comment by j-barber** [comment]
Currently our Sonar configuration uses the ["SonarWay" Quality Profile](https://sonarcloud.io/organizations/default/rules?activation=true&qprofile=java-sonar-way-45126).
![image](https://user-images.githubusercontent.com/23705265/37171717-c8f9b8be-22dc-11e8-85e5-72fbdf5d4ccd.png)

Anything failing one of the rules configured there does cause Sonar to complain.

However, there are rules we should enable that aren't enabled. For example, the rule requiring Javadoc is _not_ part of the sonar way:
https://sonarcloud.io/organizations/default/rules?languages=java&open=squid%3AUndocumentedApi&q=javadoc

And I couldn't find a rule specifying that method params be marked 'final' -- closest I've found is this rule requiring parameters to at least be read before being reassigned: 
https://sonarcloud.io/organizations/default/rules?languages=java&open=squid%3AS1226&q=S1226
Though looks like there _was_ a rule for this:
https://sbforge.org/sonar/rules/show/checkstyle:com.puppycrawl.tools.checkstyle.checks.FinalParametersCheck?layout=false

To see rules that are defined, but not part of the SonarWay, look here:
https://sonarcloud.io/organizations/default/rules?activation=false&languages=java&qprofile=java-sonar-way-45126

**Comment by j-barber** [comment]
We previously disabled checkstyle, reasoning that Sonar covers everything checkstyle might.

If we find that not to be the case, we can re-enable the checkstyle step in the Jenkins build to call checkstyle:check.

However we'll have to update the default google checks we were using, since those also don't validate that javadoc exists, nor that method params are final.

![image](https://user-images.githubusercontent.com/23705265/37172099-e2272a78-22dd-11e8-8b5b-bdfc8690a4df.png)

If we copy/update the google checks currently being used, we can get checkstyle to complain about these:
![image](https://user-images.githubusercontent.com/23705265/37172442-c6541d82-22de-11e8-94e4-f74aa0db1ddf.png)

**Comment by j-barber** [comment]
To update the quality profile we use we need to migrate out of the OpenSource organization. Since we're not admins of that org we can't update/change quality profiles or quality gates.

In my own org (where I am an admin) I have the option of creating a new Quality Profile, which I can then adjust:
![image](https://user-images.githubusercontent.com/23705265/37178939-fce4e3e6-22f1-11e8-868a-c07784017951.png)
